builder-base/scripts/download_golang.sh

```diff
@@ -27,44 +27,63 @@ RELEASE_NUMBER="$(echo $VERSION | cut -d'-' -f 2)"
-                echo "Checksum doesn't match!"
-                exit 1
-            fi
-        fi
-    done
+      if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]]; then
+        echo "Checksum doesn't match!"
+        exit 1
+      fi
+    fi
+  done
 
-    for artifact in golang-docs golang-misc golang-tests golang-src; do
-        local filename="$outputDir/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm"
-        if [ ! -f $filename ]; then
-            curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $filename --create-dirs
-            curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm.sha256 -o $filename.sha256
+  for artifact in golang-docs golang-misc golang-tests golang-src; do
+    local filename="$outputDir/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm"
+    if [ ! -f $filename ]; then
+      curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $filename --create-dirs
+      curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm.sha256 -o $filename.sha256
 
-            if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]] ; then 
-                echo "Checksum doesn't match!"
-                exit 1
-            fi
-        fi
-    done
+      if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]]; then
+        echo "Checksum doesn't match!"
+        exit 1
+      fi
+    fi
+  done
 }
 
-if [[ $ARCHITECTURE =~ "linux/amd64" ]] ; then 
-    build::go::download "${VERSION}" "$OUTPUT_DIR" "x86_64"
-fi
+function build::go::download {
+  # Set up specific go version by using go get, additional versions apart from default can be installed by calling
+  # the function again with the specific parameter.
+  local version=${1%-*}
+  local outputDir=${2}
+  local archs=${3}
+
+  for arch in ${archs/,/ }; do
+    local filename="$outputDir/${arch}/go$version.${arch/\//-}.tar.gz"
+    if [ ! -f $filename ]; then
+      curl -sSLf --retry 5 "https://go.dev/dl/go$version.${arch/\//-}.tar.gz" -o $filename --create-dirs
+    fi
+  done
+}
 
-if [[ $ARCHITECTURE =~ "linux/arm64" ]] ; then 
-    build::go::download "${VERSION}" "$OUTPUT_DIR" "aarch64"
-fi
\ No newline at end of file
+if [[ ${VERSION:2:2} -ge "21" ]]; then
+  build::go::download "${VERSION}" "$OUTPUT_DIR" "$ARCHITECTURE"
+else
+  if [[ $ARCHITECTURE =~ "linux/amd64" ]]; then
```

Comment:
id probably do something like [this](https://github.com/aws/eks-distro-build-tooling/blob/bd4d565dc6e2abeab70239def2d425ce5a7ad8f8/builder-base/scripts/install_govc.sh#L25) setting a temp var and only calling eksgo::download in one spot, but a not a huge deal.